buildpack.md

```diff
@@ -86,57 +89,108 @@ Executable: `/bin/detect <platform[AR]> <plan[E]>`, Working Dir: `<app[AR]>`
+| `<layers>/<layer>/env.launch/` | Env vars for launch (after `env`, before `profile.d`)
+| `<layers>/<layer>/env.build/`  | Env vars for subsequent buildpacks (after `env`)
+| `<layers>/<layer>/*`           | Other content for launch and/or subsequent buildpacks
+
+### Layer Types
+
+Using the [Layer Content Metadata](#layer-content-metadata-(toml)) provided by a buildpack in a `<layers>/<layer>.toml` file, the lifecycle MUST determine:
+
+- Whether the layer directory in `<layers>/<layer>/` should be available to the app.
+- Whether the layer directory in `<layers>/<layer>/` should be available to subsequent buildpacks.
+- Whether and how the layer directory in `<layers>/<layer>/` should be persisted to subsequent builds of the same OCI image.
+
+This section does not apply to the Development Setup phase, which does not generate an OCI image. 
+
+#### Launch Layers
+
+A buildpack MAY specify that a `<layers>/<layer>/` directory is a launch layer by placing `launch = true` in `<layers>/<layer>.toml`.
+
+The lifecycle MUST make all launch layers accessible to the app as defined in the [Environment](#environment) section.
+
+The lifecycle MUST include each launch layer in the built OCI image.
+The lifecycle MUST also store the Layer Content Metadata associated with each layer so that it can be recovered using the layer Diff ID.
+
+Before the next re-build:
+- The lifecycle MUST restore each `<layers>/<layer>.toml` file containing Layer Content Metadata from the previous build to the same location.
+- If a layer is cached, the lifecycle MUST also restore the corresponding `<layers>/<layer>/` directory from the previous build to the same location.
+- If a layer is not cached, the lifecycle MUST NOT restore the corresponding layer contents from any previous build.
+
+After the next re-build:
+- If a buildpack keeps `launch = true` in `<layers>/<layer>.toml` and leaves no `<layers>/<layer>/` directory, then the lifecycle MUST keep the corresponding layer from the previous build in the built OCI image.
+- If a buildpack keeps `launch = true` in `<layers>/<layer>.toml` and leaves a `<layers>/<layer>/` directory, then the lifecycle MUST replace the corresponding layer from the previous build with the directory contents.
+- If a buildpack removes `launch = true` from `<layers>/<layer>.toml` or deletes `<layers>/<layer>.toml`, then the lifecycle MUST NOT include any corresponding layer in the built OCI image.
+
+#### Build Layers
+
+A buildpack MAY specify that a `<layers>/<layer>/` directory is a build layer by placing `build = true` in `<layers>/<layer>.toml`.
+
+The lifecycle MUST make all build layers accessible to subsequent buildpacks as defined in the [Environment](#environment) section.
+
+Before the next re-build:
+- If the layer is cached, the lifecycle MAY restore the `<layers>/<layer>/` directory and Layer Content Metadata from any previous build to the same location.
```

Comment:
Just want to clarify that the use of *MAY* here is to preserve the previous cache layer behavior.  This is considered a best-effort and choosing build+cache doesn't immediately imply that the layer is preserved in a registry somewhere the way a launch+cache would be.

Reply:
Yes, although the platform can define "best effort." Also, given that platforms can choose to make this cache fully-coherent, I'd like to loosen the restriction for launch layers as well (Note: @jkutner @hone)

Reply:
I endorse loosening the restriction for launch.